Language updates

tommy9 opened the pull request:
Fixes #16 and #6

The fix for #6 also adds handling for numbers in exponent format e.g. 1.2E-10.

I also converted the textmate grammar from XML to JSON for readability. Sorry if it makes the diff for this pull request harder to read but the conversion is a separate commit, so it is possible to see the step changes for each functional change by clicking on each commit.

Apart from the 2 linked issues, I also changed:

- Handling of word boundaries around keywords and functions as opcodes weren't getting highlighted if they weren't indented
- Added all the new keywords and functions added in beebasm 1.08-1.10

simondotm commented:
This is great work @tommy9 thanks!
It's been a while since I look at this project, so I will setup some Github actions and publish a new version 👍

tommy9 commented:
That's great, thanks. 

I hope in about a month to submit a much larger pull request. I've been making progress over the last 2 months to migrate this to a language server protocol based extension. I've ported the language parser from beebasm to Typescript so it can analyse the code for issues without having to compile. I've also added function completions, go to definition and find all definitions. It is buggy and only works with single file projects so far, so there is some way to go but I'm optimistic :)

tommy9 commented:
Pushed a small update to the functions:
- added CALLSTACK$ and FILELINE$ functions as well as STR$~ variant of STR$
- separate section for functions which don't take a following open bracket (TIME$ optionally and the two above)
- stop including opening bracket in the match for functions
- changed category for functions to entity.name.function to differentiate the highlighting from regular keywords
